### PR TITLE
{chem}[foss/2023a] PySCF v2.7.0, NECI v20230620, tblis v20230422

### PR DESCRIPTION
**Pull request by PetrKralCZ**
(created using `eb --new-pr`)
resolves https://github.com/vscentrum/vsc-software-stack/issues/422
needs:
- https://github.com/easybuilders/easybuild-easyconfigs/pull/21522
- https://github.com/easybuilders/easybuild-easyconfigs/pull/21523
- https://github.com/easybuilders/easybuild-easyconfigs/pull/21524
- https://github.com/easybuilders/easybuild-easyconfigs/pull/21525
- https://github.com/easybuilders/easybuild-easyconfigs/pull/21750

### CODE REVIEW

**Comment by PetrKralCZ** [comment]
There are some disabled tests for `NECI` because they end up with a `SEGFAULT`. I tested it on a cluster with `360Gi` memory available but it didn't help. Everything else seems to work.

File: easybuild/easyconfigs/n/NECI/NECI-20230620-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+]
+
+# disable tests ending up with SEGFAULT
+preconfigopts = "sed -i '/back_spawn_excit_gen/d' '%(builddir)s/NECI_STABLE/unit_tests/CMakeLists.txt' && "
+
+# enable support for HDF5
+configopts = "-DENABLE_HDF5=ON"
+
+test_cmd = 'ctest'
+runtest = '-j'
```
Comment (boegel) [inline easybuild/easyconfigs/n/NECI/NECI-20230620-foss-2023a.eb:41]: If tests are memory-hungry, disable running them in parallel (and then maybe skipping tests that segfault is not required)

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/n/NECI/NECI-20230620-foss-2023a.eb (hunk 1 above)
Comment (PetrKralCZ) [reply easybuild/easyconfigs/n/NECI/NECI-20230620-foss-2023a.eb:41]: `runtest = '-j 1'` didn't help

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ generoso

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ jsc-zen3

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ generoso

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/n/NECI/NECI-20230620-foss-2023a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/n/NECI/NECI-20230620-foss-2023a.eb:38]: OK, it's just the tests that are very memory-hungry then, so it segfaults arise people will need to make sure they somehow make sure that enough memory is available...

**Review by boegel — APPROVED** [review]

**Review by boegel — DISMISSED** [review]
lgtm

**Comment by boegel** [comment]
I keep seeing segfaults in the NECI tests on our systems, even when *a lot* of memory is available (~185GB for 36 cores, full Skylake node on `skitty` cluster) which is weird because the test suite works just fine on `generoso` and `jsc-zen3`.

I've opened an issue upstream on this, see https://github.com/ghb24/NECI_STABLE/issues/18

We also have an old issue ourselves on this, we ran into this problem before with an old toolchain and old "version" of NECI, see https://github.com/easybuilders/easybuild-easyconfigs/issues/17164

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ generoso

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ jsc-zen3

**Comment by PetrKralCZ** [comment]
Test report by @PetrKralCZ
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node4003.donphan.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 545.23.08, Python 3.6.8
See https://gist.github.com/PetrKralCZ/42311e3a2ff2c514984f858bb653f108 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (3 easyconfigs in total)
node3164.skitty.os - Linux RHEL 9.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.9.18
See https://gist.github.com/boegel/e2cf653be9b4003cee692cc331d0b39e for a full test report.

**Comment by boegel** [comment]
Going in, thanks @PetrKralCZ!

**Pull request merged**